src/core/qgsvectorlayer.cpp

```diff
@@ -896,7 +896,10 @@ bool QgsVectorLayer::setSubsetString( const QString &subset )
   updateFields();
 
   if ( res )
+  {
+    emit configChanged();
```

Comment:
I'd probably still make a `setSubsetString()`, that's nice to have anyway :)

Reply:
@m-kuhn , OK, I was thinking about this too.

Reply:
And for extra points, add the Q_PROPERTY for subsetString on top

Reply:
And for triple points, add unit tests with signal spies for all these signals :)